### PR TITLE
[5.5] Add Model::only method

### PR DESCRIPTION
**Pull request by sebastiandedeyne**
Adds an `only` method to the `Model` class that behaves like `Request::only`, to quickly extract attributes from a model to an array.

```php
$user->only('first_name', 'last_name');
// ["first_name" => "Taylor", "last_name" => "Otwell"]
```

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php [hunk 1]
```diff
@@ -883,6 +883,23 @@ public function getOriginal($key = null, $default = null)
...
+     */
+    public function only($attributes)
+    {
+        $results = [];
+
+        foreach (is_array($attributes) ? $attributes : func_get_args() as $attribute) {
+            $results[$attribute] = $this->getAttribute($attribute);
+        }
+
+        return $results;
```
Comment (tillkruss) [inline src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php]: Why not use an array_map here?

Comment (sebastiandedeyne) [inline src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php]: Used the same style as `Request::only` https://github.com/laravel/framework/blob/78fedf79e41cc11e2608e3a06433d5b7c50962ed/src/Illuminate/Http/Concerns/InteractsWithInput.php#L144

**Comment by sebastiandedeyne** [comment]
Dot notation would be cool, but more complex since we're also dealing with accessors. I'm willing to update the PR if there's interest.

**Comment by ntzm** [comment]
This change has been rejected before: #15748

Hopefully this does get merged in though, I find myself wanting it in new projects quite a lot!

File: src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php (hunk 1 above)
Comment (ntzm) [inline src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php]: @tillkruss `array_map` won't work here because it's setting the key

**Comment by sebastiandedeyne** [comment]
Regarding the previous PR feedback

> Going to hold off on this since I feel like you could just do `Arr::only($model->toArray()...`

An upside of having this on the model is that you could call `only` on fields that are otherwise hidden.

File: src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php (hunk 1 above)
Comment (guilhermeportela) [inline src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php]: It could be done with `array_flip` or `array_combine` (which I prefer) though.

**Comment by lucasmichot** [comment]
@sebastiandedeyne, what about hidden attributes?
With your PR, they are returned without any visibility control

**Comment by calebporzio** [comment]
@sebastiandedeyne - Needed this just yesterday!  "what do you mean "->only" isn't defined???!!!"

**Comment by sebastiandedeyne** [comment]
@lucasmichot I disagree on your first point. This is the perfect escape hatch for retrieving a serialiazed version of the model *despite* the visibility attributes.

When calling `only`, you're declaring which attributes you want to extract. You can't accidentally share something with the world since it's explicit.

**Pull request merged**